### DISCUSSION TITLE
How to change time format to 24h?

### DISCUSSION THREAD

**Discussion by denisbriski** [AzuraCast/AzuraCast#4496] [Q&A]
Hello,

in the current instance of AzuraCast the time is displayed in AM/PM format.
How do I change this to 24H format?

I can't find any option regarding to this.
The timezone is set to CEST (Europe/Zagreb).
The locale on the server is set to en_GB.utf8.

**Comment by mr-denniz** [comment]
I have the same issue. Only when I switch to my native language (Dutch) the time format is changed to 24H, but I'd like my interface language to be English yet keep the 24H format.
Can an account setting be added to switch between 12H or 24H format?

**Reply by thehijacker** [reply]
I too can not change the time format to 24h. Running Azurecast in Proxmox Ubuntu 18.04 container over docker. I tried to change container locale to sl_SI (Slovenian) and restarting docker container, but still the same.

For sure I am missing something. Anyone can suggest where to look?

Thank you.

**Reply by radioindradhanush** [reply]
Go to your system's "Time & Language" settings, then click "Date & Time", then click "Language & Region", then click "Administrative language settings" which will open up a pop-up window in which you need to click "Formats" tab, and ther, under the label "Language preferences", select the desired "Short time" and "Long time" formats for 12-hour or 24-hour clock, as desired by you.

**Comment by ChrisDuffley** [comment]
I haven't seen it anywhere myself.

Christopher Duffley

Host of Mission Possible Podcast <http://www.missionpossiblepodcast.com/>

LINKS TO SUBSCRIBE:

iTunes <https://itunes.apple.com/us/podcast/mission-possible-christopher/id1103092109?mt=2> | Stitcher <http://www.stitcher.com/s?fid=88094&refid=stpr> | Google Play <https://goo.gl/app/playmusic?ibi=com.google.PlayMusic&isi=691797987&ius=googleplaymusic&link=https://play.google.com/music/m/Ik76dexklcprwnpuxbdr2s6bm3y?t=Mission_Possible_with_Christopher_Duffley> | Follow Podcast On TuneIn And Listen To Latest Episode <http://tunein.com/radio/Mission-Possible-With-Christopher-Duffley-p892505/>

LIKE/FOLLOW Me:

Facebook <http://facebook.com/christopherduffley> | Twitter <http://twitter.com/chrisduffley> | YouTube <http://youtube.com/christopherduffley> | Instagram <http://instagram.com/christopherduffley>

On Jun 21, 2022, at 07:40, thehijacker ***@***.***> wrote:


I too can not change the time format to 24h. Running Azurecast in Proxmox Ubuntu 18.04 container over docker. I tried to change container locale to sl_SI (Slovenian) and restarting docker container, but still the same.

For sure I am missing something. Anyone can suggest where to look?

Thank you.

**Reply by radioindradhanush** [reply]
Go to your system's "Time & Language" settings, then click "Date & Time", then click "Language & Region", then click "Administrative language settings" which will open up a pop-up window in which you need to click "Formats" tab, and ther, under the label "Language preferences", select the desired "Short time" and "Long time" formats for 12-hour or 24-hour clock, as desired by you.

**Comment by guiltlab** [comment]
I would also like to do this, but it is not possible AFAIK.
please vote on this suggestion to make this a feature :)
https://features.azuracast.com/suggestions/52566/ability-to-set-time-format

PS: also kinda related: https://features.azuracast.com/suggestions/292282/change-date-format

**Reply by thehijacker** [reply]
Voted in both. Thank you.

**Comment by ItsMitchh** [comment]
I'll look into this next week, my only concern is if I change to a 24 hour time, it may interact with playlists, a good suggestion nonetheless.

**Comment by radioindradhanush** [comment]
Go to your system's "Time & Language" settings, then click "Date & Time", then click "Language & Region", then click "Administrative language settings" which will open up a pop-up window in which you need to click "Formats" tab, and ther, under the label "Language preferences", select the desired "Short time" and "Long time" formats for 12-hour or 24-hour clock, as desired by you.

**Comment by Moonbase59** [comment]
No "clicking" on servers. _If_ the date/time formats in AzuraCast are taken from the Docker host’s OS, you could (on most Linuxes) use `localectl` to set the desired format for `LC_TIME` _only_, as in
```bash
localectl set-locale LC_TIME=de_DE.UTF-8
```

Be sure to check if the desired locale is installed, though:
```bash
localectl list-locales
```

After that, a restart of the Docker might be needed.

You might also succeed by adding an appropriate environment variable to your Docker. Edit `/var/azuracast/.env` and add (as example):
```bash
LC_TIME=de_DE.UTF-8
```

Time to experiment, because I can’t just now terminate my AzuraCast and test it for you. Let us know your findings.

**Reply by kick-radiomeuh** [reply]
I've tried this steps above with v0.20, and see no change.
Almost all times are in AM/PM format when my host is in fr_FR, and .env have LC_TIME=fr_FR.UTF-8 .
(after 'docker restart azuracast' ...)

Idealy, it should be a by user pref, but a global that works would be ok ;)

**Reply by BusterNeece** [reply]
@kick-radiomeuh It *is* a user preference in the profile page. Just update that and you should see 24-hour time across the system.

**Reply by Moonbase59** [reply]
Yup, just go to _My Account_ click _Edit Profile_ and there you have it—per user:

![Auswahl_405](https://github.com/AzuraCast/AzuraCast/assets/3706922/1e698c24-429b-4a50-b3f7-6a49139b5bc9)
